Flying monster tracking range

NewwindServer opened the pull request:
Flying monsters are their own category for entity activation, but, for some strange reason they are bundled with regular monsters for entity tracking. 

This is a problem because ghasts pop in very noticeably due to their size, this PR adds a way to increase their tracking range specifically to alleviate this issue.

If not set in the config, it will default to the old behavior of using monster tracking range.

NewwindServer committed:
flying monster activation range

NewwindServer committed:
Flying monster tracking range

NewwindServer committed:
include y tracking range

NewwindServer commented:
> Should probably also then extend the paper y tracking range config options.

Good catch. Fixed now

NewwindServer committed:
Remove comments

NewwindServer committed:
oops

lynxplay commented:
A question that I cannot answer right now is how sane the defaulting is.
Making these things their own category obviously means we do wanna use what people *currently* have for values.

For your spigot implementation that works fine (as long as the default val for flying monsters and monsters never diverge which, fair). The paper one is a bit more confusing as `default` generally means "what vanilla goes with", which kinda breaks here. For paper configs we have migration tooling around already so it *might* be better to register a new config version and pre-fill existing configs with their current monster value.

But I need more time to think about that which is not around given .11 releases on the 9th :sweat_smile:

NewwindServer committed:
include happy ghasts as flying monsters

They are essentially the same as ghasts, large, flying monsters, they don't currently get included because happyghast doesn't extend ghast for some reason, this fixes that
